library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -424,6 +432,7 @@ public synchronized void setUser(User user, final TokenProvider provider, Client
      * @param provider the Token Provider used to obtain the auth token for the user
      */
     public synchronized void setUser(@NotNull User user, @NotNull final TokenProvider provider) {
+        anonymousConnection = false;
         if (user == null) {
```

Comment:
shall we drop the null check here since the user arg is already decorated with `@NotNull`?

Reply:
removed